### DISCUSSION TITLE
How to remove one item listed in `:chi`?

### DISCUSSION THREAD

**Discussion by nyngwang** [stevearc/qf_helper.nvim#6] [Q&A] [answered]
I would like to "free" one of the quickfix lists I currently have. But I haven't found any docs about how to do this.

**Comment by stevearc** [comment, marked as answer]
I don't personally use the quickfix history ever, but from looking at the docs it doesn't seem like there's any way to delete a specific list from the history. There's the `setqflist([], 'f')` function, but that clears all items.

**Reply by nyngwang** [reply]
Another issue: In the document of `getqflist`, there are fields called `end_lnum`, `end_col` to enable multi-line "target". While in `setqflist` these two fields are missing. My additional question is simple: if one cannot set it how can one get it?

**Reply by stevearc** [reply]
From looking at the commit https://github.com/vim/vim/commit/6864efa59636ccede2af24e3f5f92d78d210d77b, it seems like support for `end_lnum` and `end_col` were added, but docs were only added under getqflist. You can still pass them to setqflist and it works as expected.

It is unfortunate but true that vim docs are sometimes incorrect or out of date.
This is doubly true for Neovim, since they port some-but-not-all doc changes and some-but-not-all code changes from vim, and sometimes that gets out of sync.

**Reply by nyngwang** [reply]
Thanks for your careful answer! I opened an issue there and the author made an update: https://github.com/vim/vim/issues/9676 (asked many places because of _anxiety_, while this is a happy result). (btw... while I've spent a lot of time, I found that I haven't completely understood how to facilitate these two options.)

**Reply by stevearc** [reply]
That's great that you got the docs updated!
